lib/paychex/client/companies.rb

```diff
@@ -40,12 +40,17 @@ def details_by_display_id(display_id)
           company = content[0]
           return {
             "company": company,
-            "message": company.nil? ? 'not-found' : 'found'
+            "message": 'found'
+          }
+        rescue Paychex::NotFound => e
+          return {
+            "company": nil,
+            "message": 'not found'
           }
         rescue Paychex::NoAccess => e
           return {
             "company": nil,
-            "message": 'unknown'
+            "message": 'not found'

```

Comment:
NoAccess returns `not-found` too to match the response from the previous logic of listing all the linked companies and then checking for the company in that list.